cob_twist_controller/include/cob_twist_controller/constraints/constraint_ca_impl.h

```diff
@@ -287,7 +287,8 @@ void CollisionAvoidance<T_PARAMS, PRIO>::calcPartialValues()
         {
             if (params.frame_names.end() != str_it)
             {
-                Eigen::Vector3d collision_pnt_vector = it->nearest_point_frame_vector - it->frame_vector;
+                Eigen::Vector3d collision_pnt_vector = it->nearest_point_frame_vector;
```

Comment:
Marco had to implement its own fk solver for each link such that he could compute the distances between those links. The distance was computed to the center of the object, but he needed the distance to the surface. So he had to compute the frame_vector for each mesh. This vector represents the distance from the centered point of the object to the point on the surface. By subtracting those two vectors, he obtained the real minimal distance between two links.

Since we replaced this inefficient distance computation to the moveIt distance computation, we automatically obtain the minimal distance to the surface. That's why this is valid and we don't need to subtract anything.